### ISSUE TITLE
Wedge scanner does not support hidden characters

### ISSUE THREAD

**Issue by SchrodingersGat** [inventree/inventree-app#542]
"Hidden" (non printable) chars are missed when scanning a barcode using  a "wedge" scanner (connected via BT or USB and works like a keyboard).

This is due to limitations of the flutter KeyboardListener class

**Comment by Idea-Junkie** [comment]
I got my android dev environment set up and working, going to try and see if there is any new info or if there is a way around this when I have some free time, let me know if you have any ideas as well!

**Comment by Idea-Junkie** [comment]
https://pub.dev/packages/honeywell_scanner  heres also the [Github_Link](https://github.com/luis901101/honeywell_scanner) 

Found this plugin, the reason I bring this up is because this includes the model of "scanner" I have currently. I briefly looked at it and it looked interesting however I don't know for sure if this would fix the issue, as well as supporting a variety of scanners would be difficult. 

also let me know if I need to create a seperate discussion for this or if talking about it here is good! @SchrodingersGat

**Comment by SchrodingersGat** [comment]
I believe that a "native" package may work better than the "generic" implementation which simply uses keyboard wedge mode.

The way that the barcode interface is written allows different barcode "plugins" to be used somewhat agnostically - so theoretically we could add support for the honeywell scanner and the user can select that from the dropdown list if they have a honeywell device.

As you have the device, perhaps you would be willing to implement and test this? If you come up with something clean I would be happy to add support for this so that other users can benefit too.

Refer to `/lib/barcore/barcode.dart:scanBarcode` to see how the actual scanner is selected dynamically:

https://github.com/inventree/inventree-app/blob/286daf2567c3f357127dc5168db6e348387f700f/lib/barcode/barcode.dart#L74

**Comment by Idea-Junkie** [comment]
Yes I would be willing to support this, it would be silly for me to tell you to support my small use case lol. I will probably be asking some questions here and there because dart is completely new to me, I had used python previously but even still it took me a second to understand some stuff. 

I will still try and see if I can get the keyboard wedge to work with generic external scanners(but we will see)

**Comment by Idea-Junkie** [comment]
@SchrodingersGat I did some messing around with the CT50 that I have, and actually was able to get the barcode to read and send by modifying some things in the settings. however it still did not find a match as what I did got rid of the separators altogether. 

<kbd>
<img width="762" height="604" alt="Image" src="https://github.com/user-attachments/assets/3b0ec3cc-71aa-4b06-a689-9062b94830a2" />
</kbd> 

now with this in mind the keystrokes I told the keyboard wedge to ignore were ascii "29,30,4" as per the digi-keys barcode seperators mentioned in [this article](https://forum.digikey.com/t/digikey-product-labels-decoding-digikey-barcodes/41097) 

I then soon figured out that the GS operator seems to be skipped in from the wedge? as I could narrow it down to "30,4" (RS and EOT) 

This seems to be a device or android thing as I was going to different apps with the same "profile" and it still did this, as well as here is the "Log output" from my device, where you can see the obvious spacing between the two. (top only includes GS seperator and bottom is full digi-key barcode)

## Digi-Key Barcode Scann Raw
<kbd>
<img width="720" height="214" alt="Image" src="https://github.com/user-attachments/assets/326b1845-673f-4e50-8ba5-45b9921b09ac" />
</kbd>

## Pre-made one with only GS separator
<kbd>
<img width="720" height="197" alt="Image" src="https://github.com/user-attachments/assets/c9619230-96fc-4dc2-9739-a745f3afadeb" />
</kbd>


## Conclusion so far
From what I am gathering they seem to be ignored entirely from scan to "keyboard wedge" which makes sense, but also is not the greatest, as that means there may be no way to handle the separators unless we parse the barcode differently in the app. (from my understanding) Like replacing a null value to a "|" or something like that. I think that may be the start of a idea for how we may want to handle generic barcode scanners.

However once I get a USB connection to the CT50 that I have for debugging (the connectors are 100+ USD on ebay so I will be attempting to make a USB adapter this weekend) I will start messing around with the honeywell_scanner plugin as that seems to use the "broadcast intents" that android has which means theoretically I will get the raw data that we are seeing in the logs above. Which is a good sign!
